### PR TITLE
feat: nsfw save guard

### PR DESCRIPTION
**Pull request by xhayper**

### CODE REVIEW

**Review by Bas950 — APPROVED** [review]

File: src/modules/utility/commands/presences.ts [hunk 1]
```diff
@@ -55,6 +55,17 @@ export default new DiscordCommand({
 
 		if (!dbPresence) return;
 
+		if (dbPresence.metadata.tags.includes("nsfw") && (int.channel?.type == "GUILD_TEXT" && !int.channel.nsfw)) {
+			await int.editReply({
+				embeds: [{
+					title: ":mag_right: Presences • Error",
```
Comment (EncryptedDev) [inline src/modules/utility/commands/presences.ts:61]: Title seems very strange as it’s not really an error

Comment (xhayper) [reply src/modules/utility/commands/presences.ts:61]: Oops, my bad, I copied it from older commit, what should I change it to?

**Comment by xhayper** [comment]
I did not mean to remove request for Timeraa

File: src/modules/utility/commands/presences.ts [hunk 2]
```diff
@@ -55,6 +55,17 @@ export default new DiscordCommand({
 
 		if (!dbPresence) return;
 
+		if (dbPresence.metadata.tags.includes("nsfw") && (int.channel?.type == "GUILD_TEXT" && !int.channel.nsfw)) {
```
Comment (Bas950) [inline src/modules/utility/commands/presences.ts:58]: You should probs add tags to the projection in the findOne a few lines up

**Review by Bas950 — CHANGES_REQUESTED** [review]

**Comment by Timeraa** [comment]
I think in my opinion this command should work like this:

If the presence is marked as nsfw
1. show an embed/message with a button to agree (only the command executor can see)
2. if agreed show the presence embed (only for that user)
3. if denied delete the ephemeral messages

**Comment by Timeraa** [comment]
But hmm, the issue is that we can't check for the age, but adding an nsfw channel is something that I personally don't want to do in PreMiD

**Comment by Bas950** [comment]
> I think in my opinion this command should work like this:
> 
> If the presence is marked as nsfw
> 
> 1. show an embed/message with a button to agree (only the command executor can see)
> 2. if agreed show the presence embed (only for that user)
> 3. if denied delete the ephemeral messages

I like this idea, but I wouldn't mind just redirecting them to the website either.

**Comment by xhayper** [comment]
What decided if that feature should be added is the community guidelines, not us, i gonna go read it again and see if ee can infact implement that.

**Pull request closed without merging**